bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -565,6 +572,37 @@ void writeLac(final long ledgerId, final byte[] masterKey, final long lac, ByteB
         writeAndFlush(channel, completionKey, writeLacRequest);
     }
 
+    void forceLedger(final long ledgerId, ForceLedgerCallback cb, Object ctx) {
+        if (useV2WireProtocol) {
+                LOG.error("force is not allowed with v2 protocol");
+                executor.executeOrdered(ledgerId, () -> {
+                    cb.forceLedgerComplete(BKException.Code.IllegalOpException, ledgerId, addr, ctx);
+                });
+                return;
+        }
+        final long txnId = getTxnId();
+        final CompletionKey completionKey = new V3CompletionKey(txnId,
+                                                                OperationType.FORCE_LEDGER);
+        // force is mostly like addEntry hence uses addEntryTimeout
+        completionObjects.put(completionKey,
+                              new ForceLedgerCompletion(completionKey, cb,
+                                                     ctx, ledgerId));
+
+        // Build the request
+        BKPacketHeader.Builder headerBuilder = BKPacketHeader.newBuilder()
+                .setVersion(ProtocolVersion.VERSION_THREE)
+                .setOperation(OperationType.FORCE_LEDGER)
```

Comment:
Did we make protobuf changes for this?

Reply:
There is a new RPC for FORCE_LEDGER, nothing else